source/plugins/go/src/extension/extension.go

```diff
@@ -23,66 +24,92 @@ var containerType string
+		singleton = &Extension{
+			datatypeStreamIdMap:    make(map[string]string),
+			dataCollectionSettings: make(map[string]string),
+		}
 		flbLogger.Println("Extension Instance created")
 	})
 	logger = flbLogger
 	containerType = containertype
 	return singleton
 }
 
-func (e *Extension) GetOutputStreamId(datatype string, useFromCache bool) string {
-	extensionconfiglock.Lock()
-	defer extensionconfiglock.Unlock()
-	if useFromCache && len(e.datatypeStreamIdMap) > 0 && e.datatypeStreamIdMap[datatype] != "" {
-		return e.datatypeStreamIdMap[datatype]
-	}
-	var err error
-	e.datatypeStreamIdMap, err = getDataTypeToStreamIdMapping()
-	if err != nil {
-		message := fmt.Sprintf("Error getting datatype to streamid mapping: %s", err.Error())
-		logger.Printf(message)
-	}
-	return e.datatypeStreamIdMap[datatype]
-}
-
-func getDataTypeToStreamIdMapping() (map[string]string, error) {
+func getExtensionConfigs() ([]ExtensionConfig, error) {
 	guid := uuid.New()
-	datatypeOutputStreamMap := make(map[string]string)
-
 	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerInsights", "Version": "1"}
 	jsonBytes, err := json.Marshal(taggedData)
-	// TODO: this error is unhandled
 
 	var data []byte
 	enc := codec.NewEncoderBytes(&data, new(codec.MsgpackHandle))
 	if err := enc.Encode(string(jsonBytes)); err != nil {
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
 
 	fs := &FluentSocket{}
 	fs.sockAddress = "/var/run/mdsd-ci/default_fluent.socket"
 	if containerType != "" && strings.Compare(strings.ToLower(containerType), "prometheussidecar") == 0 {
 		fs.sockAddress = fmt.Sprintf("/var/run/mdsd-%s/default_fluent.socket", containerType)
 	}
+	
 	responseBytes, err := FluentSocketWriter.writeAndRead(fs, data)
 	defer FluentSocketWriter.disconnect(fs)
 	if err != nil {
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
-	response := string(responseBytes) // TODO: why is this converted to a string then back into a []byte?
-
-	var responseObjet AgentTaggedDataResponse
-	err = json.Unmarshal([]byte(response), &responseObjet)
+	var responseObject AgentTaggedDataResponse
+	err = json.Unmarshal(responseBytes, &responseObject)
 	if err != nil {
 		logger.Printf("Error::mdsd::Failed to unmarshal config data. Error message: %s", string(err.Error()))
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
 
 	var extensionData TaggedData
-	json.Unmarshal([]byte(responseObjet.TaggedData), &extensionData)
+	json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
 
-	extensionConfigs := extensionData.ExtensionConfigs
+	return extensionData.ExtensionConfigs, nil
+}
+
+func getExtensionSettings() (map[string]map[string]interface{}, error) {
+	extensionSettings := make(map[string]map[string]interface{})
+
+	extensionConfigs, err := getExtensionConfigs()
+	if err != nil {
+		return extensionSettings, err
+	}
+	for _, extensionConfig := range extensionConfigs {

```

Comment:
we should log warn/error message if  length of extensionConfigs is greater than 1 since we are overriding the extensionSettings, right? do you know when this can have length greater than 1?

Reply:
this only has length greater than 1 if the other extensions are set together with us